api/observability/v1/output_types.go

```diff
@@ -739,6 +739,7 @@ type LokiTuningSpec struct {
 	// Compression causes data to be compressed before sending over the network.
 	//
 	// +kubebuilder:validation:Enum:=none;gzip;snappy
+	// +kubebuilder:default=none
```

Comment:
Per our larger API discussions, @xperimental has recommended not setting any defaults because of the element of "surprise" when a CR is different from that spec'd by the user once fetched from storage.  Doing this means you will do nothing but the return will be set to none.

I believe the only way to resolve this is via an initializer that evaluates and modifies the CLF based upon the type and datamodel

Reply:
Revise code again and think problem was in validation rule, added checking `!has(self.tuning.compression)`